common/transform/src/main/java/ai/h2o/mojos/deploy/common/transform/MojoFrameToContributionResponseConverter.java

```diff
@@ -0,0 +1,90 @@
+package ai.h2o.mojos.deploy.common.transform;
+
+import ai.h2o.mojos.deploy.common.rest.model.ContributionOutputGroup;
+import ai.h2o.mojos.deploy.common.rest.model.ContributionResponse;
+import ai.h2o.mojos.deploy.common.rest.model.Row;
+import ai.h2o.mojos.runtime.frame.MojoFrame;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Converts the resulting predicted {@link MojoFrame} into the API response object {@link
+ * ContributionResponse}.
+ */
+public class MojoFrameToContributionResponseConverter
+        implements Function<MojoFrame, ContributionResponse> {
+  @Override
+  public ContributionResponse apply(
+          MojoFrame shapleyMojoFrame) {
+    List<Row> outputRows = Stream.generate(Row::new).limit(shapleyMojoFrame.getNrows())
+            .collect(Collectors.toList());
+    Utils.copyResultFields(shapleyMojoFrame, outputRows);
+
+    List<String> outputFeatureNames = new ArrayList<>(
+            Arrays.asList(shapleyMojoFrame.getColumnNames()));
+
+    ContributionResponse contributionResponse = new ContributionResponse();
+    contributionResponse.setFeatures(outputFeatureNames);
+    contributionResponse.setContributionOutputGroup(new ArrayList<>());
+
+    ContributionOutputGroup contribution = new ContributionOutputGroup();
+    contribution.setContributions(outputRows);
+    // for REGRESSION and BINOMIAL models the contribution response
+    // contains only one ContributionOutputGroup object
+    contributionResponse.getContributionOutputGroup().add(contribution);
+
+    return contributionResponse;
+  }
+
+  /**
+   * Converts the resulting predicted {@link MojoFrame} into the API response object {@link
+   * ContributionResponse}.
+   */
+  public ContributionResponse apply(
+          MojoFrame shapleyMojoFrame, List<String> outputGroupNames) {
+    List<Row> outputRows = Stream.generate(Row::new).limit(shapleyMojoFrame.getNrows())
+            .collect(Collectors.toList());
+    Utils.copyResultFields(shapleyMojoFrame, outputRows);
+
+    List<String> outputFieldNames = new ArrayList<>(
+            Arrays.asList(shapleyMojoFrame.getColumnNames()));
+
+    ContributionResponse contributionResponse = new ContributionResponse();
+    contributionResponse.setContributionOutputGroup(new ArrayList<>());
+    List<String> featureNames = new ArrayList<>();
+    boolean isFirstOutputGroup = true;
+
+    for (String outputGroupName : outputGroupNames) {
```

Comment:
Output group name represents the various output classes that are present. In the example given by @mmalohlava for multinomial model - we decided to classify the contribution response based on the output classes, I was trying to avoid usage of the term 'class' in java and hence went with the term outputGroup